### ISSUE TITLE
Double Basic Auth causes 500 error

### ISSUE THREAD

**Issue by justb4** [geopython/GeoHealthCheck#329]
**Describe the bug**
Since 0.8.0 GHC supports Basic Auth (Header). But if GHC runs behind a Proxy/Webserver that already forces Basic Auth, the Auth check in GHC fails. 

**To Reproduce**
Steps to reproduce the behavior, e.g.:

1. Run GHC behind a proxy where proxy forces Basic Auth popup
2. Provide proper credentials to Proxy
3. Observe 500 Internal Server Error in browser
4. Error also seen in GHC Log

**Expected Behavior**
Expected GHC to show start screen with Login option. 

**Screenshots or Logfiles**
```
2020-07-18 03:37:05,326 - init - ERROR - Exception on / [GET]
Traceback (most recent call last):
  File "/venv/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/venv/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/venv/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/venv/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/venv/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/GeoHealthCheck/GeoHealthCheck/app.py", line 252, in home
    return render_template('home.html', response=response)
  File "/venv/lib/python3.7/site-packages/flask/templating.py", line 136, in render_template
    ctx.app.update_template_context(context)
  File "/venv/lib/python3.7/site-packages/flask/app.py", line 838, in update_template_context
    context.update(func())
  File "/venv/lib/python3.7/site-packages/flask_login/utils.py", line 368, in _user_context_processor
    return dict(current_user=_get_user())
  File "/venv/lib/python3.7/site-packages/flask_login/utils.py", line 335, in _get_user
    current_app.login_manager._load_user()
  File "/venv/lib/python3.7/site-packages/flask_login/login_manager.py", line 355, in _load_user
    return self._load_from_request(request)
  File "/venv/lib/python3.7/site-packages/flask_login/login_manager.py", line 412, in _load_from_request
    user = self.request_callback(request)
  File "/GeoHealthCheck/GeoHealthCheck/app.py", line 165, in load_user_from_request
    username, password = base64.b64decode(basic_auth_val).split(':')
TypeError: a bytes-like object is required, not 'str'
```

**Context (please complete the following information):**
 - OS: any
 - Browser any
 - Browser Version any
 - Python Version any
 - GeoHealthCheck Version: 0.8.0 and later

If running with Docker:
 - Docker installed version
 - GeoHealthCheck Docker Image version: : 0.8.0 and later

**Additional context**
Need investigation what is best strategy.

**Comment by justb4** [comment]
Suggested solution is to have a GHC var like `GHC_BASIC_AUTH_ENABLED` with a default of `True`. Whenever there is an external Basic Auth active this can be set to `False`. This does not mean that the GHC User is automatically authorized, but mainly to disable Basic Auth checking. Normal UI-based Login will still work and is always in force. Later with e.g. API Token Auth we may still have GHC-specific authorization.

**Comment by justb4** [comment]
Solved with PR #330
